src/acceptance-test/java/de/zalando/aruha/nakadi/repository/kafka/KafkaRepositoryTest.java

```diff
@@ -0,0 +1,96 @@
+package de.zalando.aruha.nakadi.repository.kafka;
+
+import de.zalando.aruha.nakadi.repository.zookeeper.ZooKeeperHolder;
+import de.zalando.aruha.nakadi.utils.TestUtils;
+import org.apache.curator.CuratorZookeeperClient;
+import org.apache.curator.framework.CuratorFramework;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.apache.kafka.common.PartitionInfo;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.arrayWithSize;
+import static org.hamcrest.Matchers.hasItem;
+import static org.hamcrest.Matchers.hasSize;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class KafkaRepositoryTest {
+
+    private static final String zookeeperURL = "localhost:2181";
+    private static final String kafkaURL = "localhost:9092";
+    private static final int defaultPartitionNum = 8;
+    private static final int defaultReplicaFactor = 1;
+
+    private KafkaRepositorySettings repositorySettings;
+    private Properties kafkaProperties;
+
+    @Before
+    public void setup() {
+        repositorySettings = createRepositorySettings();
+        kafkaProperties = createKafkaProperties();
+    }
+
+    @Test(timeout = 10000)
+    public void whenCreateTopicThenTopicIsCreated() throws InterruptedException {
+
+        // ARRANGE //
+        final String topicName = TestUtils.randomString();
+
+        final CuratorZookeeperClient zookeeperClient = mock(CuratorZookeeperClient.class);
+        when(zookeeperClient.getCurrentConnectionString()).thenReturn(zookeeperURL);
+
+        final CuratorFramework curatorFramework = mock(CuratorFramework.class);
+        when(curatorFramework.getZookeeperClient()).thenReturn(zookeeperClient);
+
+        final ZooKeeperHolder zooKeeperHolder = mock(ZooKeeperHolder.class);
+        when(zooKeeperHolder.get()).thenReturn(curatorFramework);
+
+        final KafkaRepository kafkaRepository = new KafkaRepository(zooKeeperHolder, mock(KafkaFactory.class),
+                repositorySettings);
+
+        // ACT //
+        kafkaRepository.createTopic(topicName);
+
+        // ASSERT //
+        Thread.sleep(2000);
```

Comment:
Is there any rational on why 2000 as the sleep time? I was thinking if it would be interesting to have a "retry strategy" for async assertions. So we could keep running the asserting until it passed **or** a certain timeout was reached.

Wdyt?